test/widgets/compose_box_test.dart

```diff
@@ -405,97 +407,249 @@ void main() {
       matching: find.widgetWithIcon(IconButton, icon));
 
     void checkComposeBoxParts({required bool areShown}) {
-      check(contentFieldFinder().evaluate().length).equals(areShown ? 1 : 0);
+      final inputFieldCount = inputFieldFinder().evaluate().length;
+      areShown ? check(inputFieldCount).isGreaterThan(0) : check(inputFieldCount).equals(0);
       check(attachButtonFinder(Icons.attach_file).evaluate().length).equals(areShown ? 1 : 0);
       check(attachButtonFinder(Icons.image).evaluate().length).equals(areShown ? 1 : 0);
       check(attachButtonFinder(Icons.camera_alt).evaluate().length).equals(areShown ? 1 : 0);
     }
 
-    void checkBanner({required bool isShown}) {
-      final bannerTextFinder = find.text(GlobalLocalizations.zulipLocalizations
-        .errorBannerDeactivatedDmLabel);
-      check(bannerTextFinder.evaluate().length).equals(isShown ? 1 : 0);
+    void checkBannerWithLabel(String label, {required bool isShown}) {
+      check(find.text(label).evaluate().length).equals(isShown ? 1 : 0);
     }
 
-    void checkComposeBox({required bool isShown}) {
+    void checkComposeBoxIsShown(bool isShown, {required String bannerLabel}) {
       checkComposeBoxParts(areShown: isShown);
-      checkBanner(isShown: !isShown);
+      checkBannerWithLabel(bannerLabel, isShown: !isShown);
     }
 
-    Future<void> changeUserStatus(WidgetTester tester,
-        {required User user, required bool isActive}) async {
-      await store.handleEvent(RealmUserUpdateEvent(id: 1,
-        userId: user.userId, isActive: isActive));
-      await tester.pump();
-    }
+    group('in DMs with deactivated users', () {
+      void checkComposeBox({required bool isShown}) => checkComposeBoxIsShown(isShown,
+        bannerLabel: zulipLocalizations.errorBannerDeactivatedDmLabel);
 
-    DmNarrow dmNarrowWith(User otherUser) => DmNarrow.withUser(otherUser.userId,
-      selfUserId: eg.selfUser.userId);
+      Future<void> changeUserStatus(WidgetTester tester,
+          {required User user, required bool isActive}) async {
+        await store.handleEvent(RealmUserUpdateEvent(id: 1,
+          userId: user.userId, isActive: isActive));
+        await tester.pump();
+      }
 
-    DmNarrow groupDmNarrowWith(List<User> otherUsers) => DmNarrow.withOtherUsers(
-      otherUsers.map((u) => u.userId), selfUserId: eg.selfUser.userId);
+      DmNarrow dmNarrowWith(User otherUser) => DmNarrow.withUser(otherUser.userId,
+        selfUserId: eg.selfUser.userId);
 
-    group('1:1 DMs', () {
```

Comment:
Looks like we are both moving tests to groups and adding the new tests.
The grouping part can be done in a prep commit so the diffs will be easier to read.